js/ui/codeView.js

```diff
@@ -437,12 +449,32 @@ var CodingSession = new Lang.Class({
         this._button = new WindowTrackingButton();
         this._button.connect('clicked', this._switchWindows.bind(this));
 
+        // For now, only makes sense to monitor
+        // hackable state if there is a button
+        this._setupToyAppProxy();
+
         _ensureAfterFirstFrame(actor, () => {
             Main.layoutManager.addChrome(this._button);
             this._synchronizeButton(actor.meta_window);
         });
     },
 
+    _setupToyAppProxy: function() {
+        const ToyAppProxy = Gio.DBusProxy.makeProxyWrapper(ToyAppIface);
```

Comment:
This can be called after you declare `ToyAppIface` so we don't need to create an iface wrapper every time we need to create a proxy.